kasa/kasacamprotocol.py

```diff
@@ -0,0 +1,132 @@
+"""Protocol implementation for controlling TP-Link Kasa Cam devices."""
+
+import dataclasses
+import errno
+import logging
+from base64 import b64decode, b64encode
+from dataclasses import dataclass
+from pprint import pformat as pf
+from typing import Dict
+from urllib.parse import quote
+
+import httpx
+
+# When support for cpython older than 3.11 is dropped
+# async_timeout can be replaced with asyncio.timeout
+from async_timeout import timeout as asyncio_timeout
+
+from .credentials import Credentials
+from .exceptions import SmartDeviceException
+from .json import loads as json_loads
+from .protocol import TPLinkSmartHomeProtocol
+
+_LOGGER = logging.getLogger(__name__)
+_NO_RETRY_ERRORS = {errno.EHOSTDOWN, errno.EHOSTUNREACH, errno.ECONNREFUSED}
+
+
+@dataclass
+class KasaCamCredentials(Credentials):
+    """Adapter to store credentials in format expected by Kasa Cam devices."""
+
+    def __post_init__(self):
+        """Base64 encode the password as expected by the Kasa Cam protocol."""
+        self.password: str = b64encode(self.password.encode())
+
+    @classmethod
+    def from_instance(cls, instance):
+        """Create KasaCamCredentials from Credentials."""
+        return cls(**dataclasses.asdict(instance))
```

Comment:
Instead of adding a separate class for credentials, maybe it's better to use a regular `Credentials` instance and convert the password inside `KasaCamProtocol`?

Reply:
fixed